Source/Core/VideoBackends/D3DCommon/Shader.cpp

```diff
@@ -105,8 +108,9 @@ std::optional<Shader::BinaryData> Shader::CompileShader(D3D_FEATURE_LEVEL featur
   if (FAILED(hr))
   {
     static int num_failures = 0;
-    std::string filename = StringFromFormat(
-        "%sbad_%s_%04i.txt", File::GetUserPath(D_DUMP_IDX).c_str(), target, num_failures++);
+    std::string filename = StringFromFormat("%sbad_%s_%04i_", File::GetUserPath(D_DUMP_IDX).c_str(),
+                                            target, num_failures++) +
+                           Common::scm_rev_str + "_" + g_video_backend->GetDisplayName() + ".txt";
```

Comment:
Not a big fan of using a global just to get the video backend name, when it is already known. Could you just replace this with "D3D" (and OGL/Vulkan below)?

Reply:
I'd like to differentiate D3D11 vs D3D12. Is there a better way to accomplish this?

Reply:
Oh right, I forgot that this is used for both D3D11 and D3D12. I'm not sure if there's a better way to do that... Maybe @stenzek would know?